sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -114,81 +174,260 @@ class SentryApollo3HttpInterceptor @JvmOverloads constructor(private val hub: IH
-        return request.valueForHeader(headerName)?.let {
+        return getHeader(headerName, request.headers)?.let {
             try {
                 String(Base64.decode(it, Base64.NO_WRAP))
-            } catch (e: IllegalArgumentException) {
-                hub.options.logger.log(SentryLevel.ERROR, "Error decoding internal apolloHeader $headerName", e)
+            } catch (e: Throwable) {
+                hub.options.logger.log(
+                    SentryLevel.ERROR,
+                    "Error decoding internal apolloHeader $headerName",
+                    e
+                )
                 return null
             }
         }
     }
 
-    private fun HttpRequest.valueForHeader(key: String) = headers.firstOrNull { it.name == key }?.value
+    private fun finish(
+        span: ISpan?,
+        request: HttpRequest,
+        response: HttpResponse?,
+        statusCode: Int?,
+        operationName: String?,
+        operationType: String?,
+        operationId: String?
+    ) {
+        var responseContentLength: Long? = null
+        response?.body?.buffer?.size?.ifHasValidLength {
+            responseContentLength = it
+        }
 
-    private fun finish(span: ISpan, request: HttpRequest, response: HttpResponse? = null, statusCode: Int?) {
-        if (beforeSpan != null) {
-            try {
-                val result = beforeSpan.execute(span, request, response)
-                if (result == null) {
-                    // Span is dropped
-                    span.spanContext.sampled = false
+        if (span != null) {
+            statusCode?.let {
+                span.setData("http.response.status_code", statusCode)
+            }
+            responseContentLength?.let {
+                span.setData("http.response_content_length", it)
+            }
+            if (beforeSpan != null) {
+                try {
+                    val result = beforeSpan.execute(span, request, response)
+                    if (result == null) {
+                        // Span is dropped
+                        span.spanContext.sampled = false
+                    }
+                } catch (e: Throwable) {
+                    hub.options.logger.log(
+                        SentryLevel.ERROR,
+                        "An error occurred while executing beforeSpan on ApolloInterceptor",
+                        e
+                    )
                 }
-            } catch (e: Throwable) {
-                hub.options.logger.log(SentryLevel.ERROR, "An error occurred while executing beforeSpan on ApolloInterceptor", e)
             }
+            span.finish()
         }
-        span.finish()
 
         val breadcrumb = Breadcrumb.http(request.url, request.method.name, statusCode)
 
         request.body?.contentLength.ifHasValidLength { contentLength ->
             breadcrumb.setData("request_body_size", contentLength)
         }
 
+        operationName?.let {
```

Comment:
`l`: should we prefix these with `graphql.`?

Reply:
Actually what I want is to extend https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/ adding a new `type` and `category` for crumbs specifically to `graphql`, I am checking this on the FE first, we can easily decide to change it later since `data` is not indexed anyway, if we do that, `graph.$something` is redundant because the `category` is already `graphql`.

Reply:
I see, that makes sense 👍